### PR TITLE
Feature/avmedia formats

### PR DESCRIPTION
**Pull request by 10dimensions**
- Add formats to Input Dropdown 
- Upload dynamic file type/encoding
- List Format Extensions

In progress;
- CID Input Encoding Check
- File API to distinguish audio and video

### CODE REVIEW

**Comment by gravenp** [comment]
Tested 3Dmodel, audio, image, and video object uploads successfully (front-end & Ceramic). 

Based on your previous notes on outstanding issues and the latest commit, the last outstanding issue is as follows:

- All .mp4 files are being associated to the video/mp4 file format automatically, and user cannot change it. 
- For now, we're going to require the user to manually select video or audio for mp4s as appropriate. 
- The File Format dropdown should remain as "Select a File Format" upon upload of an mp4 (or any other file extension with multiple file format options)
- File Format is a required field and the user must select from the dropdown list (filtered according to the file extension)

**Comment by gravenp** [comment]
@10dimensions I'm now able to select the encoding format of my choice, but with an MP4, the @type is still being auto-set to VideoObject. Need the type to be consistent the user's choice of encoding format. See screenshot.

Also, since we aren't filtering the File Format drop down list based on the file uploaded, we should at least sort the selections alphabetically. Let me know if you'd prefer if I create that as separate ticket.

Screen Shot 2021-11-15 at 12 23 30 PM

**Comment by 10dimensions** [comment]
@gravenp I did double check the video/audio type selection. Can you perhaps share the actual files that you were uploading? 

Yea, for alphabetical arrangement we can probably move off to a separate ticket.

File: components/gallery/GalleryFileFormat.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const galleryFileFormats = [
+    { extension: ".glb", encoding: "model/gltf-binary", type: "3DModel" },
```
Comment (gravenp) [inline components/gallery/GalleryFileFormat.js:2]: To address https://github.com/Geo-Web-Project/cadastre/issues/89, I believe these should just be listed in the following order:

{ extension: ".gif", encoding: "image/gif", type: "ImageObject" },
{ extension: ".glb", encoding: "model/gltf-binary", type: "3DModel" },
{ extension: ".jpeg", encoding: "image/jpeg", type: "ImageObject" },
{ extension: ".mp3", encoding: "audio/mpeg", type: "AudioObject" },
{ extension: ".mp4", encoding: "audio/mp4", type: "AudioObject" },
{ extension: ".mp4", encoding: "video/mp4", type: "VideoObject" },
{ extension: ".mpeg", encoding: "video/mpeg", type: "VideoObject" },
{ extension: ".png", encoding: "image/png", type: "ImageObject" },
{ extension: ".svg", encoding: "image/svg+xml", type: "ImageObject" },
{ extension: ".usdz", encoding: "model/vnd.usdz+zip", type: "3DModel" }

**Review by gravenp — CHANGES_REQUESTED** [review]
Right now, @type is being set automatically based off the file extension, but the encodingFormat is being set by the user. This allows mismatches between these two fields when they should always be consistent.

Because it's possible for a file extension to be two different @types (i.e. .mp4 can be audio or video), we are committing to manual entry method for now. We need to remove the logic that is automatically setting the @type and set this field based on what the user selects for encodingFormat. 

Even if the user selects the wrong encodingFormat, these two fields should always be consistent.

**Comment by 10dimensions** [comment]
@gravenp Types are not set upon upload now. But only from dropdown selection. 
Also, the ordering of the formats is updated

**Review by gravenp — APPROVED** [review]

**Comment by gravenp** [comment]
Tested different file formats uploads and edits. Working as intended now. Thanks @10dimensions! Merging this now.

**Pull request merged**